src/runtime/ppx_deriving_runtime.cppo.ml

```diff
@@ -14,7 +14,7 @@ type nonrec int64 = int64
 type nonrec 'a lazy_t = 'a lazy_t
 type nonrec bytes = bytes
 
-#if OCAML_VERSION >= (4, 08, 0)
+#if OCAML_VERSION >= (4, 07, 0)
```

Comment:
I'm slightly worried by this change here; but I'm not sure how we would know if it was incorrect (see the comment below); is it fine if the type-checker agress?

Note that in pre-ppxlib you added Option to the other branch, but not this one. If you do this then you have to include it here as well, as Option in 4.08.

Reply:
(You are right for `Option`: I will cherry-pick into `master` the addition of `Option` from `pre-ppxlib`, and if we agree on this PR, I will cherry-pick it as well into `pre-ppxlib`.)

I understood the comment below (which should go away, indeed, if we agree with this change) that the reason not to handle OCaml 4.07 as its successors was indeed to make the type-checker agree.

If we make `module Stdlib = Stdlib` in the implementation, we cannot make `module Stdlib = Pervasives` in the interface, but for a strange reason, the interface that is used for 4.08 and above does not type check in 4.07, basically because the following signature does not type check either (`out_channel` can be replaced by any type declared in `Stdlib`):
```ocaml
include (module type of Stdlib with
  type out_channel = Stdlib.out_channel)
```
I should admit I didn't track down why this signature does not type check in 4.07 and type check with 4.08 and above. I checked however that 4.07 still has the type equalities `out_channel = Pervasives.out_channel = Stdlib.out_channel = Stdlib.Pervasives.out_channel`.

The problem disappears with `include module type of struct include`, and I don't see which type equalities we could have lost with that construction, so I think it is fine to unify the code for 4.07 with 4.08 and above. But if you prefer to stay on the safe path and keep the former version for 4.07, I would agree with that.

Reply:
I don't understand why we need to use `include (module type of (struct include M end))` instead of `include (module type of M)` directly, do you? I remember that these two constructs are not type-checked in the same way, but not what's the difference (something something strengthening), I had the vague feeling that `struct include M end` should be weaker.

Reply:
`module type of struct include M end` gets the equalities generated by `include M`. For instance:
```ocaml
# module M = struct type t end;;
module M : sig type t end
# module type T = module type of M;;
module type T = sig type t end
# module type T = module type of struct include M end;;
module type T = sig type t = M.t end
```
It is shorter but should be equivalent to write all the equalities explicitly:
```ocaml
# module type T = (module type of M with type t = M.t);;
module type T = sig type t = M.t end
```

Reply:
(I mentioned this trick some times ago here: http://gallium.inria.fr/blog/overriding-submodules/#comment-4440770785 -- see the second example in my comment, which is shorter but should be equivalent to the solution you proposed above for overriding `Bigarray`.)

Reply:
Ah, so `module type of struct include M end` is the *stronger* variant, not the weaker one. Confusing...

With this precision in, I think that I am happy with the change.